src/pages/index.js

```diff
@@ -0,0 +1,302 @@
+    });
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal.modal_opened");
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+function closeByOverlayClick(evt) {
+  if (evt.target.classList.contains("modal_opened")) {
+    closeModal(evt.target);
+  }
+}
+
+function createCard(data) {
+  const { name, link, isLiked } = data;
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const deleteBtn = cardElement.querySelector(".card__delete-button");
+  const likeBtn = cardElement.querySelector(".card__like-button");
+
+  cardImage.src = link;
+  cardImage.alt = name;
+  cardTitle.textContent = name;
+
+  if (isLiked) {
+    likeBtn.classList.add("card__like-button_active");
+  }
+
+  likeBtn.addEventListener("click", () => {
+    if (likeBtn.classList.contains("card__like-button_active")) {
+      api
+        .removeLike(data._id)
+        .then((updatedCard) => {
+          likeBtn.classList.remove("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      api
+        .addLike(data._id)
+        .then((updatedCard) => {
+          likeBtn.classList.add("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  });
+
+  deleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  cardImage.addEventListener("click", () => {
+    previewImage.src = link;
+    previewImage.alt = name;
+    previewCaption.textContent = name;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtn = editForm.querySelector(".modal__save-btn");
+  const originalText = saveBtn.textContent;
+  saveBtn.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editNameInput.value,
+      about: editDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      resetValidation(editForm, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      saveBtn.textContent = originalText;
+    });
+}
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtn = addCardForm.querySelector(".modal__save-btn");
+  const originalText = saveBtn.textContent;
+  saveBtn.textContent = "Saving...";
+
+  const newCard = {
+    name: addCardCaptionInput.value,
+    link: addCardLinkInput.value,
+  };
+
+  api
+    .addCard(newCard)
+    .then((cardData) => {
+      cardsList.prepend(createCard(cardData));
+      closeModal(addCardModal);
+      addCardForm.reset();
+      resetValidation(addCardForm, settings);
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.

------------------------------


`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submission. It’s a **waste of resources**. You need to call `disableButton` there.


`      disableButton(saveBtn, settings);
`